jcl/src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -116,14 +116,21 @@ public boolean tryHandlePendingReference() {
  */	
 public void clear() {
 	synchronized(this) {
-		referent = null;
+		clearImpl();
 		/* change the state to cleared if it's not already cleared or enqueued */
 		if (STATE_INITIAL == state) {
```

Comment:
I believe this code should be moved to `clearImpl()` as well.  Otherwise, we could just null the field directly.

Note, `clearImpl()` can be mad synchronized as its the method that needs to keep the referent and state consistent

Reply:
done